### ISSUE TITLE
Validate `value` type is inconsistent for block field type

### ISSUE THREAD

**Issue by evelynhathaway** [payloadcms/payload#3263]
### Link to reproduction

https://github.com/evelynhathaway/reproduction-payload-block-validation

### To Reproduce

1. Use the reproduction collection
```ts
export const PostsCollection: CollectionConfig = {
  slug: "posts",
  fields: [
    {
      type: "blocks",
      blocks: [{
        slug: "text",
        fields: [
          {
            name: "Content",
            type: "text",
          },
        ],
      }],
      name: "content",
      validate: (value, {data}) => {
        console.log("Value: ", value, "Data:", data);
        if (value == null) return true;
        if (Array.isArray(value)) return "Pretend I wanted to make sure the blocks were undefined, this should error and it doesn't";
        if (value.length < 1) return "Pretend I wanted to make sure there was at least one block (ignoring mixRows of course), this should error and it doesn't";
        return true;
      }
    }
  ],
};
```
2. Open the dev tools console
3. Create a new post
4. Add a block to the blocks field
5. Click publish

### Describe the Bug

Basically, most of the time, the validation function receives a number (likely the length of the array) as the `value` of a block field, when I believe it should be an array.

---

The `value` argument of the validation function for the block field type starts out as `undefined` when the post is first created, which is desired as the field is empty by default.

When the block selector is opened, the `value` becomes `0`. I would also expect that `options.data.content`, would be exactly the same as the `value`, but it's `undefined`. **A number as the value is unexpected, and the `value` not matching the value in the document data is also unexpected.**

When a block is added, the `value` becomes `1`, and `options.data.content` becomes an array of length 1.

When the page is published, the `value` is no longer a number, and is instead identical to `options.data.content`, the expected array. This is desired, but inconsistent with how it behaves before a publication.

When the page re-renders after the publication, the `value` is back to `1` again.

I couldn't find this behavior in the documentation and it seems like a bug. It occurs on both the client side (used in this reproduction) as well as the server side.

![screenshot of the console with the above description](https://github.com/payloadcms/payload/assets/38994283/fb26ef58-f8b8-49de-b656-94b9caed56a1)

---

A naïve workaround may be to check if the value is an array before validating the field or the length of the array, however this doesn't work because it skips enough of the validations client- and server-side that an error will never appear. The field will still update with what is expected to be invalid.

See the two example error conditions in the reproduction for how this may impact someone who never tested to make sure their validations failed when expected. Fixing this bug could cause validations to start failing for the first time; however, if I understand the situation correctly, the breaking change makes validation for blocks work in the first place, which may not necessitate waiting for v2.0.

### Payload Version

1.15.2 and Master branch @ commit 8fc953605a4e15010e23b0086b1e958f7c083ccf

**Comment by JarrodMFlesch** [comment]
Hey @evelynhathaway I think this is a good change to make. In 2.0 we will be making the switch to storing the array value instead of the array length in form state. You can follow along and keep tabs on the change in this PR https://github.com/payloadcms/payload/pull/3286.

Thank you for the detailed write up!

**Comment by jmikrut** [comment]
Hey @evelynhathaway — this is now fixed in 2.0 and array / block validation functions accept the array data as expected.

👍 

Thank you!
